### PR TITLE
TS-4503: Replace MachineFatal by ink_abort

### PR DESCRIPTION
**Pull request by persiaAziz**
TS-4503: Remove the class ErrorClass and all its child classes …(for e.g., machinFatal, processFatal...).

### CODE REVIEW

**Comment by shinrich** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/747/ for details.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/643/ for details.

File: proxy/Main.cc [hunk 1]
```diff
@@ -908,7 +908,7 @@ check_fd_limit()
   if (fds_throttle > fds_limit + THROTTLE_FD_HEADROOM) {
     int new_fds_throttle = fds_limit - THROTTLE_FD_HEADROOM;
     if (new_fds_throttle < 1) {
-      MachineFatal("too few file descritors (%d) available", fds_limit);
+      ink_abort("too few file descritors (%d) available", fds_limit);
```
Comment (jpeach) [inline proxy/Main.cc]: ``Fatal`` would be better than ``ink_abort`` since it formats its message to the right log sinks before actually aborting.

**Comment by zwoop** [comment]
```
../../proxy/http/remap/UrlMapping.h:30:19: fatal error: Error.h: No such file or directory
 #include "Error.h"
```

**Comment by shinrich** [comment]
[approve ci]

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/650/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/754/ for details.

File: proxy/Main.cc (hunk 1 above)
Comment (persiaAziz) [inline proxy/Main.cc]: You are right. I have replaced the ink_aborts by Fatal.

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/761/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/657/ for details.

File: proxy/logging/LogBuffer.h [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #define LOG_BUFFER_H
 
 #include "ts/ink_platform.h"
+#include "ts/Diags.h"
```
Comment (jpeach) [inline proxy/logging/LogBuffer.h]: Is this change necessary?

**Comment by jpeach** [comment]
This looks good, I have 1 small comment.

@persiaAziz can you update the commit message to have a shorter summary and a better description of this change, see the commit message [guidelines](https://cwiki.apache.org/confluence/display/TS/CommitPolicies). Thanks.

File: proxy/logging/LogBuffer.h (hunk 2 above)
Comment (persiaAziz) [inline proxy/logging/LogBuffer.h]: Yes. I found LogBuffer complaining about not recognizing the Diags function "Note(...)" after I removed Error.h

Comment (jpeach) [inline proxy/logging/LogBuffer.h]: Better to add ``Diags.h`` to the relevant .cc files in that case.

Comment (persiaAziz) [inline proxy/logging/LogBuffer.h]: LogBuffer.h:314 calls "Note(..", that is why I have added Diags.h to this file

File: proxy/Main.cc (hunk 1 above)
Comment (persiaAziz) [inline proxy/Main.cc]: Oh I should use ink_abort here since Fatal does not call abort().  I think the original idea was to  terminate immediately at this point without memory cleanup

File: proxy/logging/LogBuffer.h (hunk 2 above)
Comment (jpeach) [inline proxy/logging/LogBuffer.h]: Ah, that's fine then.

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/773/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/669/ for details.
